packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-type-form.tsx

```diff
@@ -20,56 +24,111 @@ import {
   Tooltip,
   Typography,
 } from "@mui/material";
-import { useEffect, useState } from "react";
-import { Controller, useForm } from "react-hook-form";
-import { frontendUrl } from "@hashintel/hash-shared/environment";
-import { getPropertyTypeById } from "@hashintel/hash-subgraph/src/stdlib/element/property-type";
-import { useBlockProtocolCreatePropertyType } from "../../../../components/hooks/blockProtocolFunctions/ontology/useBlockProtocolCreatePropertyType";
+import {
+  bindDialog,
+  bindToggle,
+  PopupState,
+} from "material-ui-popup-state/hooks";
+import { ComponentProps, ReactNode, useEffect, useMemo, useState } from "react";
+import { Controller, useForm, UseFormTrigger } from "react-hook-form";
 import { useBlockProtocolGetPropertyType } from "../../../../components/hooks/blockProtocolFunctions/ontology/useBlockProtocolGetPropertyType";
+import { Modal } from "../../../../components/Modals/Modal";
 import { fa100 } from "../../../../shared/icons/pro/fa-100";
 import { faSquareCheck } from "../../../../shared/icons/pro/fa-square-check";
 import { faText } from "../../../../shared/icons/pro/fa-text";
 import { QuestionIcon } from "./question-icon";
-import { useRefetchPropertyTypes } from "./use-property-types";
 import { useRouteNamespace } from "./use-route-namespace";
+import { withHandler } from "./util";
 
 const generateInitialPropertyTypeId = (baseUri: string) =>
   addVersionToBaseUri(baseUri, 1);
 
-const propertyTypeDataTypes = [
-  {
+const propertyTypeDataTypesOptions = [
+  types.dataType.text.dataTypeId,
+  types.dataType.number.dataTypeId,
+  types.dataType.boolean.dataTypeId,
+];
+
+const propertyTypeDataTypeData = {
+  [types.dataType.text.dataTypeId]: {
     title: types.dataType.text.title,
     icon: faText,
-    dataTypeId: types.dataType.text.dataTypeId,
   },
-  {
+  [types.dataType.number.dataTypeId]: {
     title: types.dataType.number.title,
     icon: fa100,
-    dataTypeId: types.dataType.number.dataTypeId,
   },
-  {
+  [types.dataType.boolean.dataTypeId]: {
     title: types.dataType.boolean.title,
     icon: faSquareCheck,
-    dataTypeId: types.dataType.boolean.dataTypeId,
   },
-];
+};
 
-type PropertyTypeFormValues = {
+export type PropertyTypeFormValues = {
   name: string;
   description: string;
-  expectedValues: typeof propertyTypeDataTypes;
+  expectedValues: VersionedUri[];
 };
 
-export const PropertyTypeForm = ({
-  discardButtonProps,
-  initialTitle,
-  onCreatePropertyType,
+type PropertyTypeFormSubmitProps = Omit<
+  ButtonProps,
+  "size" | "variant" | "disabled" | "type" | "loading"
+>;
+
+const useTriggerValidation = (
```

Comment:
We have to trigger validation even when default values are set, to make the "submit" button not disabled. React hook form doesn't do this for us.